### PR TITLE
[SYSTEMML-834] Improve MLContext DataFrame support

### PR DESCRIPTION
**Pull request by deroneriksson**
Adds support for 4 types of DataFrame inputs and outputs via MLContext (doubles with ID column, doubles with no ID column, vectors with ID column, vectors with no ID column).

### CODE REVIEW

File: src/main/java/org/apache/sysml/api/mlcontext/Matrix.java [hunk 1]
```diff
@@ -103,16 +103,58 @@ public MatrixObject asMatrixObject() {
...
+	}
+
+	/**
+	 * Obtain the matrix as a {@code DataFrame} of doubles with no ID column
+	 * 
+	 * @return the matrix as a {@code DataFrame} of doubles with no ID column
+	 */
+	public DataFrame asDataFrameDoubleNoIDColumn() {
+		DataFrame df = MLContextConversionUtil.matrixObjectToDataFrame(matrixObject, sparkExecutionContext, false);
+		df = df.drop("ID");
```
Comment (dusenberrymw) [inline src/main/java/org/apache/sysml/api/mlcontext/Matrix.java]: I think we need to sort before dropping the "ID" column to ensure the order is correct, i.e. `df = df.sort("ID").drop("ID")`.

File: src/main/java/org/apache/sysml/api/mlcontext/Matrix.java [hunk 2]
```diff
@@ -103,16 +103,58 @@ public MatrixObject asMatrixObject() {
...
+	}
+
+	/**
+	 * Obtain the matrix as a {@code DataFrame} of vectors with no ID column
+	 * 
+	 * @return the matrix as a {@code DataFrame} of vectors with no ID column
+	 */
+	public DataFrame asDataFrameVectorNoIDColumn() {
+		DataFrame df = MLContextConversionUtil.matrixObjectToDataFrame(matrixObject, sparkExecutionContext, true);
+		df = df.drop("ID");
```
Comment (dusenberrymw) [inline src/main/java/org/apache/sysml/api/mlcontext/Matrix.java]: I think we need to sort before dropping the "ID" column to ensure the order is correct, i.e. `df = df.sort("ID").drop("ID")`.

**Comment by deroneriksson** [comment]
@dusenberrymw Added sorts before the drops

**Comment by dusenberrymw** [comment]
LGTM.

**Comment by deroneriksson** [comment]
Thank you for reviewing @dusenberrymw! I will merge.

**Pull request closed without merging**